### ISSUE TITLE
ASO/APG Goal Error

### ISSUE THREAD

**Issue by rydougherty** [ombegov/ITDB-schema#318]
Name: Ryan Dougherty
Organization: Booz Allen, supporting eCPIC PMO
Urgency: Medium, Agency Data Issue
Description of Issue: The IT Dashboard has not removed Performance Metric references to the old ASO/APG codes. This prevents Agencies from submitting only the Business Case (capitalAssetPlan) through the Annual Submission. Agencies will no longer reference old codes in the Agency Code Table in the Business Case. Because the old codes are still stored within Performance Metrics, the IT Dashboard returns an error for the old codes not appearing in the Agency Code Table. 

>  ITDB Error: 000493-BCD: The investment objectiveOrGoal (71501) is currently associated with an open performance metric (20793) and cannot be deactivated/changed

**Comment by elliotyoon202** [comment]
ITD-1943 (Internal Tracking)

**Comment by courtneycorwin** [comment]
Hi all,

As I mentioned in ticket #305 - we are no longer seeing this error message.  However, we submit our business cases in two parts, so that could be the difference.

**Comment by jeremiah-tcg** [comment]
After reviewing the error message you received, it appears that this is a valid error message for the given data. It appears you are trying to updateObjectiveorGoal but it is not allowing it because there are open performance metrics already associated with the objective/goal. On 8/22, the goal 011APG18024  had already been associated with a performance metric. 

The old codes are not the issue here. We are keeping the performance metrics with the old codes, so the performance metrics don't all have to be re-added, but the agencies must update the objective/goal for all existing performance metrics. 

In other words, to do this Agencies must:
1) Add the new objectives/goals using updateInvestment (they may use the old agencyObjectiveOrGoalId or not)
2) Update the performance metrics, updating performanceObjectiveOrGoal to a new code

**Comment by courtneycorwin** [comment]
Hi All, 

I am pretty sure this is linked to issue #305, so I thought I would post the same comment I made over there over here - I wouldn't mind closing #305 and only working on this ticket so that the information is kept all in once place. 

I've sent you three files:

1.	Input_to_OMB – this is the file sent to OMB.  The file was sent as updateInvestment and the ASO/APG in question was also sent as an update because of its presence in the Output_from_OMB file.

2.	Output_from_OMB – this is OMB’s response to getInvestment – you’ll see there is one active ObjectiveorGoal with an 11-character ID valid for BY20.  At some point, the business case for the investment must have actually been submitted to the UAT site vs. validated.

3.	Errors_from_OMB – these are the errors returned based on the input file.  The error message returned here is “000493-BCD: The investment objectiveOrGoal (45542) is currently associated with an open performance metric (24004) and cannot be deactivated/changed”.  However, as you’ll see in the output and input files, there is no objectiveOrGoal with an ID of 45542 (a valid BY19 goal that is not valid for BY20).  That Goal ID is still associated with a Performance Metric because we submit the details separately.  So, the error message is, essentially, referencing an objectiveOrGoal that no longer exists.  

Thanks for helping to resolve this!
Courtney

**Comment by jeremiah-tcg** [comment]
During our CPIC Vendor Meeting on September 6, Ryan shared that this no longer appears to be an issue. We are therefore closing this ticket.
